pyccel/codegen/printing/ccode.py

```diff
@@ -488,36 +486,56 @@ def _print_PythonLen(self, expr):
                     PYCCEL_RESTRICTION_TODO,
                     symbol = expr, severity='fatal')
 
+    def _print_Header(self, expr):
+        return ''
+
     def _print_ModuleHeader(self, expr):
         name = expr.module.name
         # TODO: Add classes and interfaces
         funcs = '\n'.join('{};'.format(self.function_signature(f)) for f in expr.module.funcs)
 
+        global_variables = ''.join(['extern '+self._print(d) for d in expr.module.declarations if not d.variable.is_private])
+
         # Print imports last to be sure that all additional_imports have been collected
         imports = [*expr.module.imports, *map(Import, self._additional_imports)]
         imports = ''.join(self._print(i) for i in imports)
 
         return ('#ifndef {name}_H\n'
                 '#define {name}_H\n\n'
                 '{imports}\n'
+                '{variables}\n'
                 #'{classes}\n'
                 '{funcs}\n'
                 #'{interfaces}\n'
                 '#endif // {name}_H\n').format(
                         name    = name.upper(),
                         imports = imports,
+                        variables = global_variables,
                         funcs   = funcs)
 
     def _print_Module(self, expr):
+        self._current_module = expr.name
         body    = ''.join(self._print(i) for i in expr.body)
 
+        global_variables = ''.join([self._print(d) for d in expr.declarations])
+
+        if expr.program:
```

Comment:
can we move this the buttom before return, and add the result to the code, the same of what we did in `fcode.py`
```python
# ...
if expr.program:
    code += self._print(expr.program)
return code
```

Reply:
I think it needs to be before the printing of the imports so we collect any `additional_imports` although, looking at `_print_Program` it seems that it prints these itself. I will fix that

Reply:
Do you want me to add a comment indicating this? Or is the comment on the next line:
```
# Print imports last to be sure that all additional_imports have been collected
```
sufficient?

Reply:
yes, this can help for better understanding.

Reply:
Fixed 39e59eb5